pyvista/core/filters/structured_grid.py

```diff
@@ -125,34 +125,39 @@ def concatenate(self, other, axis, tolerance=0.0):
 
         """
         if axis > 2:
-            raise RuntimeError('Concatenation axis must be <= 2.')
+            msg = 'Concatenation axis must be <= 2.'
+            raise RuntimeError(msg)
 
         # check dimensions are compatible
         for i, (dim1, dim2) in enumerate(zip(self.dimensions, other.dimensions)):  # type: ignore[attr-defined]
             if i == axis:
                 continue
             if dim1 != dim2:
-                raise ValueError(
+                msg = (
                     f'StructuredGrids with dimensions {self.dimensions} and {other.dimensions} '  # type: ignore[attr-defined]
-                    'are not compatible.',
+                    'are not compatible.'
                 )
+                raise ValueError(msg)
 
         # check point/cell variables are the same
-        if not set(self.point_data.keys()) == set(other.point_data.keys()):  # type: ignore[attr-defined]
-            raise RuntimeError('Grid to concatenate has different point array names.')
-        if not set(self.cell_data.keys()) == set(other.cell_data.keys()):  # type: ignore[attr-defined]
-            raise RuntimeError('Grid to concatenate has different cell array names.')
+        if set(self.point_data.keys()) != set(other.point_data.keys()):  # type: ignore[attr-defined]
```

Comment:
Weird, applying the fix `EM` on the code below the `if` triggered the `SIM201` [rule](https://docs.astral.sh/ruff/rules/negate-equal-op/#negate-equal-op-sim201) . Maybe a ruff issue ...
I saw this behavior in other places.